lib/_http_outgoing.js

```diff
@@ -296,21 +296,20 @@ function _storeHeader(firstLine, headers) {
     header: firstLine
   };
 
-  var key;
-  if (headers === this[outHeadersKey]) {
-    for (key in headers) {
-      const entry = headers[key];
-      processHeader(this, state, entry[0], entry[1], false);
-    }
-  } else if (Array.isArray(headers)) {
-    for (var i = 0; i < headers.length; i++) {
-      const entry = headers[i];
-      processHeader(this, state, entry[0], entry[1], true);
-    }
-  } else if (headers) {
-    for (key in headers) {
-      if (hasOwnProperty(headers, key)) {
-        processHeader(this, state, key, headers[key], true);
+  if (headers) {
+    if (headers === this[outHeadersKey]) {
+      for (const entry of Object.values(headers)) {
+        processHeader(this, state, entry[0], entry[1], false);
+      }
+    } else if (Array.isArray(headers)) {
+      for (const entry of headers) {
+        processHeader(this, state, entry[0], entry[1], true);
+      }
+    } else {
+      for (const key in headers) {
+        if (hasOwnProperty(headers, key)) {
```

Comment:
This part could now be combined with `if (headers === this[outHeadersKey])`. The former was probably written in a performance sensitive way and now the functionality is identical (before, the `hasOwnProperty()` check could be skipped because we were able to tell that the headers were not tampered with. `Object.values()` only checks own properties, so it's the same as calling `hasOwnProperty()`.

Reply:
Looks not viable, the fifth parameter is different.

Reply:
Oh, I missed that one!

Reply:
Can you approve this pr if there is no further questions.

Reply:
I'll give a LG as soon as I checking the performance of the two cases. So far I just abstain and if someone else is fine to land it, that's also good for me.